### PR TITLE
Add UoE papers on deep learning over code

### PR DESCRIPTION
**Pull request by ChrisCummins**
Signed-off-by: Chris Cummins <chrisc.101@gmail.com>

Please fill the `[ ]` with an `x` to make sure that you read and followed the checklist

- [x] This addition is about Machine Learning __on Code__ specifically. General links should be added to [Awesome Machine Learning](https://github.com/josephmisiti/awesome-machine-learning) instead.
- [x] I have checked that the proposed addition is not already in the awesome list.
- [x] I have signed-off my commits as detailed [here](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md).
- [x] The links follow the link template:

        * [Title](URL) - Description.
    or

        * [Title](URL) - Description [Code](link to code).
- [x] The addition is correctly classified as a paper, blog post, talk, software or dataset.
- [x] The commit message must start with a verb (`Add …`, `Deprecate …`) and end without a dot. For exemple:

        Deprecate Theano software

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ A curated list of awesome research papers, datasets and software projects devote
 
 #### Program Synthesis and Induction
 
+* <img src="badges/12-pages-beginner-brightgreen.svg" alt="11-pages" align="top"> [Compiler Fuzzing through Deep Learning](https://chriscummins.cc/pub/2018-issta.pdf) - Chris Cummins, Pavlos Petoumenos, Alastair Murray, Hugh Leather, ISSTA 2018
```
Comment (m09) [inline README.md]: This one would better fit the [Program Repair and Bug Detection](https://github.com/src-d/awesome-machine-learning-on-source-code#program-repair-and-bug-detection) section :)

Also:
```suggestion
* <img src="badges/11-pages-beginner-brightgreen.svg" alt="11-pages" align="top"> [Compiler Fuzzing through Deep Learning](https://chriscummins.cc/pub/2018-issta.pdf) - Chris Cummins, Pavlos Petoumenos, Alastair Murray, Hugh Leather, ISSTA 2018
```

File: README.md [hunk 2]
```diff
@@ -217,6 +219,7 @@ Code Vectors: Understanding Programs Through Embedded Abstracted Symbolic Traces
 
 #### Code Optimization
 
+* <img src="badges/12-pages-gray.svg" alt="12-pages" align="top"> [End-to-end Deep Learning of Optimization Heuristics](https://chriscummins.cc/pub/2017-pact.pdf) - Chris Cummins, Pavlos Petoumenos, Zheng Wang, Hugh Leather, PACT 2017
```
Comment (m09) [inline README.md]:
```suggestion
* <img src="badges/14-pages-gray.svg" alt="14-pages" align="top"> [End-to-end Deep Learning of Optimization Heuristics](https://chriscummins.cc/pub/2017-pact.pdf) - Chris Cummins, Pavlos Petoumenos, Zheng Wang, Hugh Leather, PACT 2017
```

File: README.md [hunk 3]
```diff
@@ -72,6 +72,8 @@ A curated list of awesome research papers, datasets and software projects devote
 
 #### Program Synthesis and Induction
 
+* <img src="badges/12-pages-beginner-brightgreen.svg" alt="11-pages" align="top"> [Compiler Fuzzing through Deep Learning](https://chriscummins.cc/pub/2018-issta.pdf) - Chris Cummins, Pavlos Petoumenos, Alastair Murray, Hugh Leather, ISSTA 2018
+* <img src="badges/12-pages-beginner-brightgreen.svg" alt="12-pages" align="top"> [Synthesizing Benchmarks for Predictive Modeling](https://chriscummins.cc/pub/2017-cgo.pdf) - Chris Cummins, Pavlos Petoumenos, Zheng Wang, Hugh Leather, CGO 2017
```
Comment (m09) [inline README.md]: We sort our paper chronologically. If you could move this one a few lines down that'd be perfect! Also:
```suggestion
* <img src="badges/14-pages-beginner-brightgreen.svg" alt="14-pages" align="top"> [Synthesizing Benchmarks for Predictive Modeling](https://chriscummins.cc/pub/2017-cgo.pdf) - Chris Cummins, Pavlos Petoumenos, Zheng Wang, Hugh Leather, CGO 2017
```

**Review by m09 — CHANGES_REQUESTED** [review]
Thanks @ChrisCummins for contributing your papers, they look extremely relevant :+1:

I added a few minor modifications we require before merging in the review below.

As a side note, we run a [reading club](https://github.com/src-d/reading-club) online every 2 weeks on fridays (4PM CEST, 1 hour). We'd be delighted to go through one of your papers with you. Let me know if you'd be willing to join :smile_cat:

**Comment by m09** [comment]
Sorry for the closed PR, I rebased your PR and fixed the review myself since we changed the README a bit recently and I pushed the wrong branch to your master :man_facepalming: I opened a new PR with your commit in #177.

**Comment by ChrisCummins** [comment]
Hi @m09, thanks for the review, and apologies for the shameless self-promotion. 😄

I'd be delighted to join your reading group! My schedule's pretty hectic at the moment but I should be able to e-attend them starting in late Sept.

It looks like you've done the merge, but let me know if there's anything I can do at this end.

Cheers,
Chris

**Comment by m09** [comment]
Hi @ChrisCummins, that's great :tada: If that fits your schedule, we'd be glad to book the September 20th session with one of your papers :upside_down_face:

If you'd rather figure out the details by mail, you can reach me at hugo@sourced.tech.

Cheers

**Comment by ChrisCummins** [comment]
September 20th works for me - you're on! :)

I'll shoot you an email for the details.

Cheers,
Chris

**Pull request closed without merging**